Content.Shared/Cloning/CloningPodComponent.cs

```diff
@@ -67,6 +71,50 @@ public sealed partial class CloningPodComponent : Component
 
     [ViewVariables]
     public EntityUid? ConnectedConsole;
+
+    /// <summary>
+    ///     Tracks whether a Cloner is actively cloning someone.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    public bool ActivelyCloning;
+
+    #region Metempsychosis
+    /// <summary>
+    ///     Controls whether a cloning machine performs the Metempsychosis functions, EG: Is this a Cloner or a Metem Machine?
+    ///     Metempsychosis refers to the metaphysical process of Reincarnation
+    /// </summary>
+    [DataField]
+    public bool DoMetempsychosis;
```

Comment:
It actually functions pretty much exactly the same as it did before, and unfortunately there doesn't actually exist any reasonable alternatives due to the limitations of HumanoidAppearanceSystem. The underlying issue why I can't do it any other way is that the underlying HumanoidAppearanceSystem's functions are non-destructive, non-polymorphic, cosmetic-only. Essentially, for Metempsychosis to work, the standard cloning code can't be allowed to spawn an entity in the first place, it(CloningSystem) *must* make the decision on if it's going to Clone, or Reincarnate, before it spawns the entity.

Say I'm cloning a Harpy. And I do it first by spawning a harpy entity. If I assemble an Oni Profile, and then load the Oni Profile onto the Harpy via LoadProfile, the Harpy clone will outwardly look like an Oni, and will have every single cosmetic feature of the new HumanoidCharacterProfile. But if I VV it and look at the components, this new "Oni" is going to have a HarpySingerComponent, a Harpy's FixtureComponent, a Harpy's PhysicsComponent, a Harpy's Damage Container, Inventory, DamageResistance, hell the Oni is even going to have a pair of AvianLungs. It will look like an Oni, talk like an Oni, but inside it's secretly a Harpy. 

That being said there ARE still places in this code where I can insert Events to modify the process! It's just that events cannot be used to determine spawning behavior in this way.

Reply:
Anyway, my own reasoning is that if it's impossible to actually decouple the two systems because of the limitations of a different, 10x as large and 100x as interconnected system out of the scope of this PR, that the old Metem component didn't really need to exist. A single system shouldn't be totally dependent on two components that by all rights could be combined. In order for another system to modify cloning, it shouldn't need 2 or 3 TryComps for components related to the CloningSystem, it really should only need a single TryComp that contains all the relevant information. What I think I'm actually going to do is to insert an event before the code makes the decision on whether the machine is going to Clone or Reincarnate, and optionally allow other systems to tell it, "Nope! Reincarnate me instead!", or "Always clone me". And this event should likely be all the way at the beginning of TryClone.

Reply:
I'm also probably going to replace the hardcoded check for Uncloneable with said event, and make Uncloneable work instead by listening to the event and cancelling it.